pkgs/build-support/build-dotnet-module/default.nix

```diff
@@ -0,0 +1,127 @@
+, nugetDeps ? null
+
+# Libraries that need to be available at runtime should be passed through this.
+# These get wrapped into `LD_LIBRARY_PATH`.
+, buildInputs ? []
+# The dotnet runtime to use.
+, dotnet-runtime ? dotnetCorePackages.net_5_0
+# The dotnet SDK to use.
+, dotnet-sdk ? dotnetCorePackages.sdk_5_0
+, ... } @ args:
+
+assert projectFile == null -> throw "Defining the `projectFile` attribute is required. This is usually an `.csproj`, or `.sln` file.";
+
+# TODO: Automatically generate a dependency file when a lockfile is present.
+# This file is unfortunately almost never present, as Microsoft recommands not to push this in upstream repositories.
+assert nugetDeps == null -> throw "Defining the `nugetDeps` attribute is required, as to lock the NuGet dependencies. This file can be generated using the `nuget-to-nix` tool.";
+
+let
+  _nugetDeps = linkFarmFromDrvs "${name}-nuget-deps" (import nugetDeps {
+    fetchNuGet = { name, version, sha256 }: fetchurl {
+      name = "nuget-${name}-${version}.nupkg";
+      url = "https://www.nuget.org/api/v2/package/${name}/${version}";
+      inherit sha256;
+    };
+  });
+
+  package = stdenv.mkDerivation (args // {
+    nativeBuildInputs = args.nativeBuildInputs or [] ++ [ dotnet-sdk dotnetPackages.Nuget cacert makeWrapper ];
+
+    # Stripping breaks the executable
+    dontStrip = true;
+
+    configurePhase = args.configurePhase or ''
+      runHook preConfigure
+
+      export HOME=$(mktemp -d)
+      export DOTNET_CLI_TELEMETRY_OPTOUT=1
+      export DOTNET_NOLOGO=1
+
+      nuget sources Add -Name nixos -Source "$PWD/nixos"
+      nuget init "${_nugetDeps}" "$PWD/nixos"
+
+      # This is required due to https://github.com/NuGet/Home/issues/4413.
+      mkdir -p $HOME/.nuget/NuGet
+      cp $HOME/.config/NuGet/NuGet.Config $HOME/.nuget/NuGet
+
+      dotnet restore --source "$PWD/nixos" ${projectFile} ''${dotnetRestoreFlags} ''${dotnetFlags}
+
+      runHook postConfigure
+    '';
+
+    buildPhase = args.buildPhase or ''
+      runHook preBuild
+
+      dotnet build ${projectFile} \
+        --no-restore \
+        --configuration Release \
+        -p:Version=${args.version} \
+        ''${dotnetBuildFlags} \
+        ''${dotnetFlags}
+
+      runHook postBuild
+    '';
+
+    installPhase = args.installPhase or ''
+      runHook preInstall
+
+      dotnet publish ${projectFile} \
+        --no-build \
+        --configuration Release \
+        --output $out/lib/${args.pname} \
+        ''${dotnetInstallFlags} \
+        ''${dotnetFlags}
+    '' + (if executables != null then ''
+      for executable in ''${executables}; do
+        execPath="$out/lib/${args.pname}/$executable"
+
+        if [[ -f "$execPath" ]] && [[ -x "$execPath" ]]; then
+          makeWrapper "$execPath" "$out/bin/$(basename "$executable")" \
+            --set DOTNET_ROOT "${dotnet-runtime}" \
+            --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
+            ''${gappsWrapperArgs[@]} \
```

Comment:
If this is not passed, the argument simply gets ignored. I'm not sure of any other way to do this, without double wrapping when applications do need `wrapGappsHook`.